# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/OffscreenCanvas.h
# Commit: db2347ee8147
# Full Hash: db2347ee81479323d8b188df53a9cb3c0ad6c6e7
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 09:10:34
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 1. Rename CanvasRenderingContextHelper::GetContext to GetOrCreateContext. r=jgilbert
#   
#   This patch contains no functional changes but sets us up for refactoring
#   in a later part in this series. It also helps silence a bunch of
#   compiler warnings.
# ==============================================================================

diff -r 3dde5ddb65e5 -r db2347ee8147 dom/canvas/OffscreenCanvas.h
--- a/dom/canvas/OffscreenCanvas.h	Thu Dec 09 19:25:23 2021 +0000
+++ b/dom/canvas/OffscreenCanvas.h	Thu Dec 09 19:25:24 2021 +0000
@@ -103,6 +103,10 @@
     }
   }
 
+  already_AddRefed<nsISupports> GetContext(
+      JSContext* aCx, const nsAString& aContextId,
+      JS::Handle<JS::Value> aContextOptions, ErrorResult& aRv);
+
   already_AddRefed<ImageBitmap> TransferToImageBitmap(ErrorResult& aRv);
 
   already_AddRefed<Promise> ToBlob(JSContext* aCx, const nsAString& aType,
@@ -136,10 +140,6 @@
   virtual already_AddRefed<nsICanvasRenderingContextInternal> CreateContext(
       CanvasContextType aContextType) override;
 
-  virtual already_AddRefed<nsISupports> GetContext(
-      JSContext* aCx, const nsAString& aContextId,
-      JS::Handle<JS::Value> aContextOptions, ErrorResult& aRv) override;
-
   void SetNeutered() { mNeutered = true; }
 
   bool IsNeutered() const { return mNeutered; }